internal/web/handlers/webhandl.go

```diff
@@ -0,0 +1,88 @@
+package webhandl
+
+import (
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/shulganew/shear.git/internal/app/config"
+	utils "github.com/shulganew/shear.git/internal/core"
+	"github.com/shulganew/shear.git/internal/storage"
+)
+
+// hadler for  GET and POST  hor and log urls
+
+type URLHandler struct {
+	storage storage.URLSetGet
+}
+
+func (u *URLHandler) SetMapStorage(storage storage.URLSetGet) {
+	u.storage = storage
+}
+
+func (u *URLHandler) GetStorage() storage.URLSetGet {
+	return u.storage
+}
+
+// GET and redirect by shortUrl
+func (u *URLHandler) GetURL(res http.ResponseWriter, req *http.Request) {
+
+	shortURL := chi.URLParam(req, "id")
+
+	//get long Url from storage
+	longURL, exist := u.storage.GetLongURL(shortURL)
+
+	//set content type
+	res.Header().Add("Content-Type", "text/plain")
+	log.Println("Redirect to: ", longURL)
+
+	if exist {
+		res.Header().Set("Location", longURL.String())
+		//set status code 307
+		res.WriteHeader(http.StatusTemporaryRedirect)
+	} else {
+		res.WriteHeader(http.StatusNotFound)
+	}
```

Comment:
```suggestion
	if exist {
		res.Header().Set("Location", longURL.String())
		//set status code 307
		res.WriteHeader(http.StatusTemporaryRedirect)
		
		return
	} 
	
	res.WriteHeader(http.StatusNotFound)
	
```
лишние else - не стоит использовать - приходится читать два условия вместо одного.